_posts/2020-06-04-App Service Continuous Deployment for Windows Containers with GitHub Actions.md

```diff
@@ -0,0 +1,403 @@
+
+## Create Resources
+
+Create the following resources. You will need information from each resource that will be used in the file and stored in your secrets. Create your choice of registry (Azure Container Registry or Docker Hub) first since you will need information from there before you can create your App Service.
+
+1. Azure Container Registry or a Docker Hub container registry
+1. App Service (Web App for Container)
+1. Azure SQL Database (Optional)
+
+## Create a Service Principal
+
+Our workflow will use a Service Principal to authenticate with Azure when deploying the container to App Service. A service principal is an Active Directory Identity created for use with automation scenarios, such as GitHub Actions.
+
+1. Run the following command in Azure CLI in powershell to get the credentials needed to run the login action.  The output of this command will be a collection of key value pairs that you'll need to add to your GitHub secrets.
+
+    ```shell
+    az ad sp create-for-rbac --name "<appservice-name>" --role contributor \
+                                --scopes /subscriptions/{subscription-id}/resourceGroups/{resource-group} \
+                                --sdk-auth
+    ```
+
+1. Copy the output into your GitHub secrets to use as your `AZURE_CREDENTIALS` secret.
+
+    ```json
+      {
+        "clientId": "<GUID>",
+        "clientSecret": "<GUID>",
+        "subscriptionId": "<GUID>",
+        "tenantId": "<GUID>" 
+      }
+    ```
+
+## Secure Secrets
+
+Since we are using sensitive information that you don't want others to access, we will use GitHub secrets to protect our information. Create a secret by following the directions [here](https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets).  Add the github secrets variables below with your own secrets appropriate from each resource.  If you are not using Docker Hub, ignore the DOCKERHUB prefixed parameters and if you are not using ACR, ignore the REGISTRY prefixed variables. 
+
+- `APP_NAME` = your web apps name
+- `AZURE_CREDENTIALS` = your service principal output
+- `IMAGE_NAME` = name of your image that will upload to your registry
+- `REGISTRY_USERNAME` = your registry username
+- `REGISTRY_PASSWORD` = your registry password
+- `AZURE_SQL_CONNECTION_STRING` = database connection string
+- `DATABASE_SERVER_NAME` = name of your server
+- `DOCKERHUB_REGISTRY_NAME` = your docker hub registry name
+- `DOCKERHUB_USERNAME` = your docker hub user name
+- `DOCKERHUB_PASSWORD` = your docker hub password
+
+## The Dockerfile
+
+Before we can start on our GitHub Action workflow, we also need to make sure our dockerfile is in order.  The sample(s) below are setup to work with the sample app linked above. If you need to adjust your dockerfile or create your own, remember to test building it locally first or it will not work when you run the action. 
```

Comment:
This sentence confuses me as a reader, because the opener said that the samples were already configured for GitHub Actions, but now I have to make sure the Dockerfile is in order. 

I think this should be re-worded to tell the reader that you're just explaining the contents. Right now it reads like I need to compare this dockerfile and the one in the repo to make changes.

Reply:
```suggestion
The samples below explain the associated Dockerfiles for the .NET Framework and .NET Core sample applications linked above.  If creating your own application, use the appropriate Dockerfile below and replace the directory paths to match your application.  
```